code/mission/missionmessage.cpp

```diff
@@ -728,17 +720,35 @@ void messages_init()
 
 	if ( !table_read ) {
 		Default_command_persona = -1;
-		
-		try
-		{
-			parse_msgtbl();
-			table_read = 1;
-		}
-		catch (const parse::ParseException& e)
-		{
-			mprintf(("TABLES: Unable to parse '%s'!  Error message = %s.\n", "messages.tbl", e.what()));
-			return;
-		}
+
+		// speed things up a little by setting the capacities for the message vectors to roughly the FS2 amounts
+		Messages.reserve(500);
+		Message_waves.reserve(300);
+		Message_avis.reserve(30);
+
+		Num_messages = 0;
+
+		// Add built-in generic filenames
+		Generic_message_filenames.clear();
+		Generic_message_filenames.push_back("none");
+		Generic_message_filenames.push_back("cuevoice");
+		Generic_message_filenames.push_back("cue_voice");
+		Generic_message_filenames.push_back("emptymsg");
+		Generic_message_filenames.push_back("generic");
+		Generic_message_filenames.push_back("msgstart");
+
+		// first parse the default table
+		parse_msgtbl("messages.tbl");
+
+		// parse any modular tables
+		parse_modular_table("*-msg.tbm", parse_msgtbl);
+
+		// save the number of builtin message things -- make initing between missions easier
+		Num_builtin_messages = Num_messages;
+		Num_builtin_avis = Num_message_avis;
+		Num_builtin_waves = Num_message_waves;
+
+		table_read = 1;
 	}
```

Comment:
Exceptions will no longer be thrown?

Reply:
try/catch exists within `parse_msgtbl()`. I didn't think we needed it here also because any parsing errors will happen within that method.